servicetalk-http-netty/src/testFixtures/java/io/servicetalk/http/netty/AsyncContextHttpFilterVerifier.java

```diff
@@ -0,0 +1,205 @@
+import io.servicetalk.http.api.StreamingHttpRequest;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.api.StreamingHttpResponseFactory;
+import io.servicetalk.http.api.StreamingHttpService;
+import io.servicetalk.http.api.StreamingHttpServiceFilter;
+import io.servicetalk.http.api.StreamingHttpServiceFilterFactory;
+import io.servicetalk.http.utils.BeforeFinallyHttpOperator;
+import io.servicetalk.transport.api.HostAndPort;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.InetSocketAddress;
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.http.api.HttpSerializationProviders.textDeserializer;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.http.netty.HttpClients.forSingleAddress;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class AsyncContextHttpFilterVerifier {
+
+    private static final AsyncContextMap.Key<String> K1 = AsyncContextMap.Key.newKey("k1");
+    private static final AsyncContextMap.Key<String> K2 = AsyncContextMap.Key.newKey("k2");
+    private static final AsyncContextMap.Key<String> K3 = AsyncContextMap.Key.newKey("k3");
+    private static final AsyncContextMap.Key<String> K4 = AsyncContextMap.Key.newKey("k4");
+
+    private static final String V1 = "v1";
+    private static final String V2 = "v2";
+    private static final String V3 = "v3";
+    private static final String V4 = "v4";
+
+    private AsyncContextHttpFilterVerifier() {
+    }
+
+    public static void verifyServerFilterAsyncContextVisibility(final StreamingHttpServiceFilterFactory filter)
+            throws Exception {
+        final BlockingQueue<Throwable> errors = new LinkedBlockingDeque<>();
+
+        final StreamingHttpServiceFilterFactory filters = asyncContextAssertionFilter(errors).append(filter);
+        final StreamingHttpService service = filters.create(asyncContextRequestHandler(errors));
+
+        final InetSocketAddress listenAddress =
+                (InetSocketAddress) HttpServers.forPort(0).listenStreamingAndAwait(service).listenAddress();
+
+        final BlockingHttpClient client = forSingleAddress(HostAndPort.of(listenAddress)).buildBlocking();
+        final HttpRequest request = client.post("/test")
+                      .payloadBody("Hello World", textSerializer());
+
+        client.request(request);
+        assertEmpty(errors);
+    }
+
+    private static StreamingHttpService asyncContextRequestHandler(final BlockingQueue<Throwable> errorQueue) {
+        return (ctx, request, respFactory) -> {
+            AsyncContext.put(K1, V1);
+            return request.payloadBody(textDeserializer()).firstOrError().map(it -> {
```

Comment:
It's unknown how OS will cut frames. Safer to use `collect(...)` here instead of `firstOrError()` in case the payload body will be split into multiple buffers.